test/e2e/action_test.go

```diff
@@ -280,6 +281,75 @@ var _ = Describe("Action", func() {
 			assertOutputTypeInstancesInActionStatus(ctx, engineClient, action.Name, And(ContainElements(uploadedTIOutput), HaveLen(1)))
 		})
 
+		It("should propagate context provider to storage backend", func() {
```

Comment:
The diff between the `"should pick Implementation A/B` and this test is only with the context (it's almost 1:1 copy-paste) and IMO it unnecessary to have a dedicated test case for that as this can be easily incorporated into A/B as there were already adjusted exactly to this. You only need to get rid of creating Helm storage backend and replace it with real store as stated in `TODO(https://github.com/capactio/capact/issues/634):` comments + add `context`. 

Having an additional test will slow down our pipelines, that the reason why it's better to reduce what we can.

Reply:
Fixed, as suggested I replaced Helm storage backend and used test storage + add context. It is better :)